src/node_buffer.cc

```diff
@@ -1091,6 +1091,46 @@ void IndexOfNumber(const FunctionCallbackInfo<Value>& args) {
                                 : -1);
 }
 
+void Swap16(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args.This());
+  SPREAD_ARG(args.This(), ts_obj);
+
+  if (ts_obj_length % sizeof(uint16_t) != 0) {
+    return env->ThrowRangeError(
+      "Buffer length must be a multiple of 16-bits");
+  }
```

Comment:
Have you considered dropping these tests and just using the ones you wrote in JS?